File: scala/bubble_sort.scala
def bubble_sort(input: Array[Int]): Array[Int] = {
    val limit = input.size - 1
    for (a <- 1 to limit) {
      for (b <- limit to a by -1) {
        if (input(b) < input(b - 1)) {
          val x = input(b)
          input(b) = input(b - 1)
          input(b - 1) = x
        }
      }
    }
    input
}

File: python/bubble_sort.py
#!/usr/bin/env python


def bubble_sort(_array):
    x = len(_array) - 1
    count = 0
    for i in range(x):
        count += 1
        print('{}: {}'.format(count, _array))
        for j in range(x, i, -1):
            if _array[j - 1] > _array[j]:
                _array[j], _array[j - 1] = _array[j - 1], _array[j]
    return _array


if __name__ == '__main__':
    import random
    buff = [random.randint(0, 100) for i in range(10)]
    print('Data: {}'.format(buff))
    bubble_sort(buff)
    print('Result: {}'.format(buff))
